skbio/alignment/_alignment.py

```diff
@@ -72,13 +72,16 @@ def __init__(self, seqs):
         self._data = seqs
         self._id_to_index = {}
         for i, seq in enumerate(self._data):
-            id = seq.id
-            if id in self:
+            if 'id' not in seq.metadata:
```

Comment:
@jairideout and I worked on this together. Should be resolved as of the latest commit.